core/domain/state_domain.py

```diff
@@ -2366,6 +2563,17 @@ def validate(self, exp_param_specs_dict, allow_null_interaction):
                 raise utils.ValidationError(
                     'Found a duplicate content id %s' % feedback_content_id)
             content_id_list.append(feedback_content_id)
+
+            for rule_spec in answer_group.rule_specs:
+                for input_name in rule_spec.inputs:
+                    rule_input = rule_spec.inputs[input_name]
+                    if isinstance(
```

Comment:
If using primitive types, just check for the existence of a content_id key. Ditto below.